src/tests/class_tests/openms/source/AASequence_test.cpp

```diff
@@ -1161,6 +1165,44 @@ START_SECTION([EXTRA] testing terminal modifications)
 }
 END_SECTION
 
+START_SECTION([EXTRA] multithreaded example)
+{
+  static ResidueDB* rdb = ResidueDB::getInstance();
+
+  // All measurements are best of three (wall time, Linux, 8 threads)
+  //
+  // Serial execution of code:
+  // 5e4 iterations -> 5.42 seconds
+  //
+  // Parallel execution of code:
+  // original code:
+  // 5e4 iterations -> 4.01 seconds with boost::shared_mutex
+  // 5e4 iterations -> 8.05 seconds with std::mutex
+  //
+  // refactored code that throws exception outside block:
+  // 5e4 iterations -> 8.55 seconds with omp critical
+  // 5e4 iterations -> 8.59 seconds with std::mutex
+  // 5e4 iterations -> 3.94 seconds with boost::shared_mutex
+  int nr_iterations (5e4), test (0);
```

Comment:
> how digestions and conversion to a AASequence object performs

@timosachsenberg does this answer your question?

Reply:
partly... also interested in the speed drop compared to the non-thread safe code .
E.g. parallel context but no proper locking

Reply:
> partly... also interested in the speed drop compared to the non-thread safe code .

that comparison wont work because you get segfaults and uncaught Exceptions all around (e.g. code trying to add Modifications to the DB but iterators becoming invalid midway through the code etc...). There is a reason we call it *thread-safe* code :-)

Reply:
I worked around the thread safety in SimpleSearchEngine by registering all Residues in a single-threaded context. So I am a bit curious how this now affects speed ;). But I can do it myself later.